### ISSUE TITLE
[BUG] Deserialization error calling `request_guid()` on `wasmcloud:extras` (wasmcloud 0.18)

### ISSUE THREAD

**Issue by StuartHarris** [wasmCloud/wasmCloud#175]
**Describe the bug**
When calling `request_guid()` on the `wasmcloud:extras` provider, we get an error `Host error: Failed to de-serialize: IO error while reading marker: failed to fill whole buffer`.

**To Reproduce**

```rust
let guid: String = extras::default()
  .request_guid()
  .unwrap_or_else(|e| format!("{:?}", e));

assert_eq!(
  guid.as_str(),
  r#"Error(HostError("Failed to de-serialize: IO error while reading marker: failed to fill whole buffer"))"#
);
```

**Expected behavior**
A guid should be returned

**Environment (please complete the following information)**

- OS: RPi 4B
- Architecture: arm64
- Shell: bash
- wasmcloud version: `0.18.0`

**Additional context**
I've tried this with `wasmcloud-actor-extras = {version = "0.1.2", features = ["guest"]}`

**Comment by StuartHarris** [comment]
I'm guessing that the fix is to drop `GeneratorRequest` and `GeneratorResult` in favor of individual args and return types? 
If so, I can submit a PR shortly.

**Comment by StuartHarris** [comment]
But it looks like that would be a breaking change, meaning that wasmcloud host wouldn't support actors that use extras earlier than v0.1.2. It could be implemented in a non-breaking way, e.g. try to deserialize into e.g. `RequestRandomArgs` first and then `GeneratorRequest` second, but it feels a bit clunky. Are there any discussions under way around contract versioning that I might be missing?

**Comment by brooksmtownsend** [comment]
Thank you for the bug report @StuartHarris, it appears our extras contract was only bumped a patch version when it should've been bumped a minor version for breaking the API. As you can see with https://github.com/wasmCloud/examples/tree/main/extras, the `0.1.1` version of the extras contract dealt with a `GeneratorResult` as the return type for all operations.

So, I think we're going to have to either bump the extras contract to `0.2.0` if it works, but I'm fairly sure that msgpack will fail to deserialize the single value type (string, u32) as a return type. We may want to simply revert the extras interface to the previous version where it returned the `GeneratorResult`

**Comment by brooksmtownsend** [comment]
Submitted a fix for this in #230 , better late than never! 😆
